nilearn/regions/region_extractor.py

```diff
@@ -0,0 +1,294 @@
+        A method used to segment/seperate the regions.
+        If 'connected_components', each component/region in the image is extracted
+        automatically by assigning unique labels based upon their features.
+        If 'local_regions', each component/region in the image is extracted based upon their
+        local peak max value giving each region a unique identity and then using random walker
+        to increase the robustness in the regions seperation.
+
+    peak_local_smooth: scalar, default is 6mm
+        a value in mm which is used to smooth an image to locate seed points.
+
+    mask_img: Nifti-like image/object, default is None
+        If given, mask image is applied directly on the input data 'maps_img'.
+        For example, if interested to consider only one particular brain region
+        then mask image to that particular brain region should be provided.
+        If none, masking will not be applied.
+
+    Returns
+    -------
+    regions_extracted: a 4D Nifti-like image
+        contains the images of segmented regions each 3D image is a
+        seperate brain activated region or atlas region.
+
+    index_of_each_map: a numpy array
+        an array of list of indices where each index value denotes the indentity
+        of each extracted region belonging to the family of brain maps.
+    """
+    all_regions_imgs = []
+    index_of_each_map = []
+    maps_img = check_niimg(maps_img, atleast_4d=True)
+    maps = maps_img.get_data()
+    affine = maps_img.get_affine()
+
+    extract_methods = ['connected_components', 'local_regions']
+    if extract_type not in extract_methods:
+        message = ("'extract_type' should be given "
+                   "either of these {0}").format(extract_methods)
+        raise ValueError(message)
+
+    if mask_img is not None:
+        if not _check_same_fov(maps_img, mask_img):
+            mask_img = resample_img(mask_img,
+                                    target_affine=maps_img.get_affine(),
+                                    target_shape=maps_img.shape[:3],
+                                    interpolation="nearest")
+            mask_data, _ = masking._load_mask_img(mask_img)
+            # Set as 0 to the values which are outside of the mask
+            maps[mask_data == 0.] = 0.
+
+    for index in range(maps.shape[-1]):
+        regions = []
+        map_ = maps[..., index]
+        # Mark the seeds using random walker
+        if extract_type == 'local_regions':
+            smooth_map = _smooth_array(map_, affine, peak_local_smooth)
+            seeds = peak_local_max(smooth_map, indices=False,
+                                   exclude_border=False)
+            seeds_label, seeds_id = label(seeds)
+            # Assign -1 to values which are 0. to indicate to ignore
+            seeds_label[map_ == 0.] = -1
+            try:
+                rw_maps = random_walker(map_, seeds_label, mode='cg_mg')
+            except Exception as e:
```

Comment:
@AlexandreAbraham Here, ```'cg_mg'``` has a dependency called ```pyamg```. By itself, it is downgrading to ```mode='cg'``` since we don't have installed ```pyamg```. The reason which I am not ```try```ing ```mode='cg'```. Do you think should I still ```try``` 'cg'